EVSL_1.1.1/INC/evsl.h

```diff
@@ -1,101 +1,140 @@
-extern "C" {
-#endif
+#define EVSL_PI 3.14159265358979323846
+
+/** for comparing floating-point numbers (double)
+ *     for "a >= b", compare "a + EVSL_DBL_EPS_MULT * DBL_EPSILON >= b"
+ *     for "a <= b", compare "a - EVSL_DBL_EPS_MULT * DBL_EPSILON >= b"
+ */
+#define EVSL_DBL_EPS_MULT 10
+
+/*!
+  \def max(x,y)
+  Computes the maximum of \a x and \a y.
+*/
+#define evsl_max(a, b) ((a) > (b) ? (a) : (b))
+
+/*!
+  \def min(x,y)
+  Computes the minimum of \a x and \a y.
+*/
+#define evsl_min(a, b) ((a) < (b) ? (a) : (b))
+
+/* memory management */
+/**
+ * @brief A malloc wrapper which provides basic error checking
+ * @param count Number of elements to be allocated
+ * @param type  Type of elements to be allocated
+ */
+#define evsl_Malloc(count, type) \
+(\
+ /* printf("[%s:%d] MALLOC %ld bytes\n", __FILE__,__LINE__, (size_t)(sizeof(type) * (count))) ,*/ \
+ (type *) _evsl_Malloc( (size_t)(sizeof(type) * (count)) ) \
+)
+
+/**
+ * @brief A calloc wrapper which provides basic error checking
+ * @param count Number of elements to be allocated
+ * @param type Type of elements to be allocated
+ */
+#define evsl_Calloc(count, type) \
+(\
+ /* printf("[%s:%d] CALLOC %ld bytes\n", __FILE__,__LINE__, (size_t)(sizeof(type) * (count))) ,*/ \
+ (type *) _evsl_Calloc( (size_t)(count), (size_t)sizeof(type) ) \
+)
+
+/**
+ * @brief A realloc wrapper which provides basic error checking
+ * @param count Number of elements to be allocated
+ * @param type Type of elements to be allocated
+ */
+#define evsl_Realloc(ptr, count, type) \
+(\
+ /* printf("[%s:%d] ReALLOC %ld bytes\n", __FILE__,__LINE__, (size_t)(sizeof(type) * (count))) ,*/ \
+ (type *) _evsl_Realloc( (void *)ptr, (size_t)(sizeof(type) * (count)) ) \
+)
+
+/**
+ * @brief A free wrapper which sets NULL
+ * @param ptr Pointer to a memory block previously allocated with evsl_Malloc, evsl_Calloc or evsl_Realloc
+ */
+#define evsl_Free(ptr) \
+(\
+ _evsl_Free((void *)ptr), \
+ ptr = NULL \
+)
 
 /*- - - - - - - - - cheblanNr.c */
-int ChebLanNr(double *intv, int maxit, double tol, double *vinit,
-              polparams *pol, int *nevOut, double **lamo, double **Wo,
-              double **reso, FILE *fstats);
+int ChebLanNr(double *intv, int maxit, double tol, double *vinit, polparams *pol, int *nevOut, double **lamo, double **Wo, double **reso, FILE *fstats);
 
 /*- - - - - - - - - cheblanTr.c */
-int ChebLanTr(int lanm, int nev, double *intv, int maxit, double tol,
-              double *vinit, polparams *pol, int *nev2, double **vals,
-              double **W, double **resW, FILE *fstats);
+int ChebLanTr(int lanm, int nev, double *intv, int maxit, double tol, double *vinit, polparams *pol, int *nev2, double **vals, double **W, double **resW, FILE *fstats);
 
 /*- - - - - - - - - chebpoly.c */
-//
 void set_pol_def(polparams *pol);
-//
 int find_pol(double *intv, polparams *pol);
-//
 void free_pol(polparams *pol);
-
-int ChebAv(polparams *pol, double *v, double *y, double *w);
```

Comment:
I had included this line so that programs using EVSL could use it, but as you've recommented it here that may have been a mistake on my part. Is there another function you would recommend for use instead?